cmd/slideshow.go

```diff
@@ -17,37 +17,81 @@ package cmd
 import (
 	"os"
 
+	"io/fs"
+
+	"github.com/rs/zerolog/log"
 	"github.com/spf13/cobra"
+
+	"path/filepath"
+	"strings"
 )
 
+type share struct {
+	files []string
+}
+
 // slideshowCmd represents the slideshow command
 var slideshowCmd = &cobra.Command{
 	Use:   "slideshow file1 file2 ...",
 	Short: "Play a slideshow of photos",
 	Run: func(cmd *cobra.Command, args []string) {
 		if len(args) == 0 {
-			exit("requires files to play in slideshow")
+			exit("requires files (or directories) to play in slideshow")
 		}
+
+		s := &share{}
+
 		for _, arg := range args {
 			if fileInfo, err := os.Stat(arg); err != nil {
-				exit("unable to find %q: %v", arg, err)
+				log.Warn().Msgf("unable to find %q: %v", arg, err)
 			} else if fileInfo.Mode().IsDir() {
-				exit("%q is a directory", arg)
+				log.Debug().Msgf("%q is a directory", arg)
+
+				// recursively find files in directory
+				// TODO: this will consume large amounts of memory as it will hold references to each file (media item) to be served
+				filepath.WalkDir(arg, s.getFilesRecursively)
+			} else {
+				s.files = append(s.files, arg)
 			}
 		}
-		app, err := castApplication(cmd, args)
+
+		app, err := castApplication(cmd, s.files)
 		if err != nil {
 			exit("unable to get cast application: %v", err)
 		}
 
 		duration, _ := cmd.Flags().GetInt("duration")
 		repeat, _ := cmd.Flags().GetBool("repeat")
-		if err := app.Slideshow(args, duration, repeat); err != nil {
+		if err := app.Slideshow(s.files, duration, repeat); err != nil {
 			exit("unable to play slideshow on cast application: %v", err)
 		}
 	},
 }
 
+func (s *share) getFilesRecursively(path string, d fs.DirEntry, err error) error {
+	if err != nil {
+		return err
+	}
+
+	fileInfo, err := os.Stat(path)
+	if err != nil {
+		log.Warn().Msgf("Error checking file: %v | %v", path, err)
+		return nil
+	}
+
+	if !fileInfo.Mode().IsDir() {
+		filename := path
+
+		if strings.HasSuffix(strings.ToLower(path), "jpg") {
```

Comment:
Do you mind updating this to be a list of all supported filetypes if we're going to filter like this: https://github.com/vishen/go-chromecast/blob/master/application/application.go#L714-L722?

If you wanted, feel free to add a flag to filter only specific image types.

Reply:
I updated this to use the list of file extensions in the other file.

Thanks,

Walter